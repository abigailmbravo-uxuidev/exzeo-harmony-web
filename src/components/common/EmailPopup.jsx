import React, { PropTypes } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { reduxForm, Form, propTypes } from 'redux-form';
import TextField from '../Form/inputs/TextField';
import * as cgActions from '../../actions/cgActions';
import * as appStateActions from '../../actions/appStateActions';

const EmailPopup = ({ submitting, handleSubmit, primaryButtonHandler, secondaryButtonHandler }) => (
  <div className="email-modal active" role="article">
    <div className="survey-wrapper">
      <div className="contact-message">
        <div className="card card-email">
          <Form className={'fade-in'} id="SendEmail" onSubmit={handleSubmit(primaryButtonHandler)} noValidate>
            <div className="card-header">
              <h4><i className="fa fa-share-alt" /> Share Quote</h4>
            </div>
            <div className="card-block">
              <TextField type="text" name={'name'} value styleName={'name'} label={'Name'} validations={['required']} />
              <TextField type="text" name={'emailAddr'} styleName={'emailAddress'} label={'Email Address'} validations={['required', 'email']} />
            </div>
            <div className="card-footer">
              <button className="btn btn-secondary" onClick={secondaryButtonHandler} type="button">Cancel</button>
              <button className="btn btn-primary" type="submit" disabled={submitting}>Send Email</button>
            </div>
          </Form>
        </div>
      </div>
    </div>
  </div>
);

EmailPopup.propTypes = {
  ...propTypes,
  primaryButtonHandler: PropTypes.func,
  secondaryButtonHandler: PropTypes.func
};

// ------------------------------------------------
// redux mapping
// ------------------------------------------------
const mapStateToProps = state => ({
  tasks: state.cg,
  appState: state.appState
});

const mapDispatchToProps = dispatch => ({
  actions: {
    cgActions: bindActionCreators(cgActions, dispatch),
    appStateActions: bindActionCreators(appStateActions, dispatch)
  }
});

// ------------------------------------------------
// wire up redux form with the redux connect
// ------------------------------------------------
export default connect(mapStateToProps, mapDispatchToProps)(reduxForm({
  form: 'SendEmail'
})(EmailPopup));

