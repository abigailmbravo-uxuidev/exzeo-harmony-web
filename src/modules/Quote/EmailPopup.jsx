import React from 'react';
import PropTypes from 'prop-types';
import { reduxForm, Field } from 'redux-form';
import { Loader, Input, Button, validation } from '@exzeo/core-ui';

const EmailPopup = ({
  submitting,
  handleSubmit,
  handleCancel,
}) => (
    <div className="email-modal modal active" role="article">
      <div className="survey-wrapper">
        <div className="contact-message">
          <div className="card card-email">
            {submitting &&
              <Loader />
            }
            <div className="fade-in" id="SendEmail">
              <div className="card-header" data-test="modal-header">
                <h4><i className="fa fa-share-alt" /> Share Quote</h4>
              </div>
              <div className="card-block">
                <Field
                  name="name"
                  styleName="text"
                  component={Input}
                  label="Name"
                  validate={validation.isRequired}
                  dataTest="name"
                />

                <Field
                  name="emailAddr"
                  styleName="text"
                  component={Input}
                  label="Email Address"
                  validate={[validation.isRequired, validation.isEmail]}
                  dataTest="emailAddr"
                />

              </div>
              <div className="card-footer">
                <Button
                  className={Button.constants.classNames.secondary}
                  onClick={handleCancel}
                  data-test="modal-cancel"
                >Cancel</Button>
                <Button
                  className={Button.constants.classNames.primary}
                  onClick={handleSubmit}
                  disabled={submitting}
                  data-test="modal-submit"
                >Send Email</Button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );

EmailPopup.propTypes = {
  primaryButtonHandler: PropTypes.func,
  secondaryButtonHandler: PropTypes.func
};

export default reduxForm({
  form: 'SendEmail'
})(EmailPopup);
