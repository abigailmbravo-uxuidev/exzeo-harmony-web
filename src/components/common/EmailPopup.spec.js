import React from 'react';
import { shallow } from 'enzyme';
import EmailPopup from './EmailPopup';

describe('EmailPopup', () => {
  it('Should render when no props are passed in', () => {
    const wrapper = shallow(<EmailPopup
      showEmailPopup={false}
      primaryButtonHandler={function () { }}
      secondaryButtonHandler={function () { }}
    />);

    expect(wrapper).to.exist;
  });
});
