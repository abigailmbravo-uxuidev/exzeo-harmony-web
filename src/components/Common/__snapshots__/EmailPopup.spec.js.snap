exports[`test renders without crashing 1`] = `
<div
  className="email-modal active"
  role="article">
  <div
    className="survey-wrapper">
    <div
      className="contact-message">
      <div
        className="card card-email">
        <form
          className="fade-in"
          id="SendEmail"
          noValidate={true}
          onSubmit={false}>
          <div
            className="card-header">
            <h4>
              <i
                className="fa fa-share-alt" />
               Share Quote
            </h4>
          </div>
          <div
            className="card-block">
            <div
              className="form-group name name">
              <label
                htmlFor="name">
                Name
                   
                
              </label>
              <input
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="text"
                value="" />
            </div>
            <div
              className="form-group emailAddress emailAddr">
              <label
                htmlFor="emailAddr">
                Email Address
                   
                
              </label>
              <input
                name="emailAddr"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="text"
                value="" />
            </div>
          </div>
          <div
            className="card-footer">
            <button
              className="btn btn-secondary"
              onClick={[Function]}
              type="button">
              Cancel
            </button>
            <button
              className="btn btn-primary"
              disabled={false}
              type="submit">
              Send Email
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
